integration_tests/conftest.py

```diff
@@ -9,6 +9,26 @@
 REPO_ROOT = "/tests/git_wrapper"  # Local copy
 
 
+def clean_up_ongoing_operations(repo):
```

Comment:
I would suggest passing the operation to clean up 'am', 'rebase' and then calling the correct abort from that point instead of using fall through logic.

Reply:
While I am all for making code cleaner, I would strong advocate NOT making such a fix if it means making it herder to write tests - it is often hard enough to convince contributors to add them, we should keep that bar as low as possible, imo.  I am sure we can decompose this in some way to clean it up, but please, just not in a way that makes testing more onerous.

Reply:
Good point! What I have in mind is referencing an additional (existing) fixture argument in the test def, so not particularly onerous but something to remember when writing the test or as reviewers. It seems to work so I'll run with it and we can discuss in the next review iteration once it's clearer what that looks like. Personally I find addCleanup()/addFinalizer()-type construct more natural and easier to understand but they only seems to work within fixtures, not within tests directly.